soccer/Processor.cpp

```diff
@@ -295,10 +305,26 @@ void Processor::run() {
             log->CopyFrom(packet->wrapper);
 
             curStatus.lastVisionTime = packet->receivedTime;
+
+            if (packet->wrapper.has_geometry()) {
+                // DEMO: Test out field sizes
+                const SSL_GeometryFieldSize fieldSize = packet->wrapper.geometry().field();
+                // FIXME - Account for network latency
+
+                if (fieldSize.field_length() != 0 &&
+                    // Compare floats with a small threshold
+                    (currentDimensions->Length() - (fieldSize.field_length() / 1000.0f)) > 0.001f) {
+                    // Set the changed field dimensions to the current ones
+                    decodeGeometryPacket(&fieldSize);
```

Comment:
Actually since `decodeGeometryPacket()` only sets it if decoding is successful, maybe keep it as-is, but rename it to reflect the fact that it's updating current dimensions in addition to doing the decoding.